### PR TITLE
Major refactor to break out .run method and create new helper class

### PR DESCRIPTION
**Pull request by lauraGgit**
* [x] Make `.run()` just a wrapper for `getCards()` and `cardRecordFunctions()` that resolves once all the card promises are done
* [x] Create new `cardRecordFunction(card)` method to handle the recording of each card
* [x] Create new `decideCommentType()` method to allow for more extensible handling of end of board cards
* [x] break out `CardRecorder` into two classes, one for the more recording, updating, and asynchronous processes, and one for the synchronous date processes.
* [ ] passing tests for refactor.

### CODE REVIEW

**Comment by mgwalker** [comment]
Sorry for the long delay, @lauraGgit.  Is this ready for review?  I should have some cycles tomorrow or Friday to look at it properly!

**Comment by lauraGgit** [comment]
@mgwalker ready for review & thoughts on the pending tests.

File: app/cardrecord.js [hunk 1]
```diff
@@ -88,6 +36,49 @@ class CardRecorder extends MyTrello {
         return deferred.promise;
     }
 
+    cardRecordFunctions(card){
```
Comment (mgwalker) [inline app/cardrecord.js]: Ticky comment: this function name is throwing me for a loop.  Does this name describe its functionality?

**Comment by mgwalker** [comment]
LGTM!

The one thing that catches my attention is some inconsistency in whitespace, brackets, etc.  I think it would be good if we added a linter to catch these kinds of things.  The front-end guild recently adopted the airbnb eslint style, and Steven and I started using it a couple of months ago.  The initial lint is a heavy lift, but thereafter it's pretty easy.  I'll take a stab at it and see what happens!

I will merge this and then take a look at the tests in another branch.

**Pull request merged**